nemo/collections/nlp/models/nlp_model.py

```diff
@@ -38,6 +41,36 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
         self.bert_model = None  # Pretrained BERT encoder
         self.set_world_size(trainer)
 
+    @rank_zero_only
+    def register_bert_model(self):
+        # check if there is an encoder, warn if not
+        if self.bert_model is None:
+            raise ValueError('Instantiate self.bert_model before registering it.')
+        else:
+            # get encoder config
+            if isinstance(self.bert_model, BertEncoder):
+                # HuggingFace Transformer Config
+                self.bert_model.config.to_json_file('nn_config.json')  # name requested by jarvis team
+                self.register_artifact('nn_config.json', 'nn_config.json')
+            elif isinstance(self.bert_model, MegatronBertEncoder):
+                config_for_json = OmegaConf.to_container(self.bert_model.config)
+                with open('nn_config.json', 'w', encoding='utf-8') as f:
+                    f.write(json.dumps(config_for_json, indent=2, sort_keys=True) + '\n')
+                self.register_artifact('nn_config.json', 'nn_config.json')
+            else:
+                logging.info(
+                    f'Registering BERT model: {self.bert_model} is not yet supported. Please override this method.'
+                )
+
+    def _setup_tokenizer(self, cfg: DictConfig):
```

Comment:
@ericharper as you are refactoring this piece, can you please take a look at this document:
https://docs.google.com/document/d/1vBBufqQE8_dtT3_6ZTUQwZdMCBqtsyslGN2ceucAdYo/edit#

is it also possible to extract e.g. the tokenizer vocabulary and add it as yet another artifact?

Reply:
Yes, that is a part of this PR.
 
I took what @ekmb did and lifted it to NLP Model so the tokenizer should be a part of the .nemo file. Right now the vocab file is not being saved, I'll look into it.

Reply:
@tkornuta-nvidia I'm not getting the vocab from the tokenizer and then saving it as json. Is that what's needed? I can't really tell from the Google doc.

Reply:
ehehe that generally makes two of us ;) still, happy that we got it, as till yesterday everything down there was _elusive_ ;)

I guess we will make few iterations on this, when JARVIS folks will learn what do they really want.

What bothers me is that we will need to update all the existing .nemo files, as we do not want to retrain every model once again. I hope once EFF will be officially released, we will work on a tool that will also help with this, similarly to the current `eff-nemo-convert`...